tests/function-controller/pkg/apirule/apirule.go

```diff
@@ -85,3 +91,71 @@ func (a *APIRule) Delete() error {
 
 	return nil
 }
+
+func (a *APIRule) get() (*apiruleTypes.APIRule, error) {
+	u, err := a.resCli.Get(a.name)
+	if err != nil {
+		return &apiruleTypes.APIRule{}, errors.Wrapf(err, "while getting ApiRule %s in namespace %s", a.name, a.namespace)
+	}
+
+	apirule := &apiruleTypes.APIRule{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, apirule)
+	if err != nil {
+		return &apiruleTypes.APIRule{}, err
+	}
+
+	return apirule, nil
+}
+
+func (a *APIRule) WaitForStatusRunning() error {
+	apirule, err := a.get()
+	if err != nil {
+		return err
+	}
+
+	if isStateReady(apirule.Status) {
+		a.log.Logf("%s is ready:\n%v", a.name, apirule)
+		return nil
+	}
```

Comment:
this prevents subtle bug in which apirule is already ready, so you get no events in `watchtools.Until` function, which just hangs for infinity